### ISSUE TITLE
Wrong generated values when regexp owns boundary matchers (eg ^ $)

### ISSUE THREAD

**Issue by tnmtechnologies** [benas/random-beans#315]
The random-beans-validation dependency is set.

The declaration is as follow :
```java
@JsonProperty("supportedFeatures")
@Pattern(regexp="^[A-Fa-f0-9]*$")
public String getSupportedFeatures() {
    return supportedFeatures;
}
```

The result is that the generated values always start with a ^ and ends with a $.
Example of a generated value :
` "supportedFeatures": "^d44b8dc8b6bbCd2AF56Bd$"`

This is wrong because the characters ^and $ are boundary matchers.
They are not allowed characters for the value.

**Comment by benas** [comment]
Hi @tnmtechnologies 

Thank you for reporting this issue! Indeed, that's obviously incorrect.
Random Beans relies on [java-faker](https://github.com/DiUS/java-faker) to generate random strings according to a given regular expression, which in turn relies on [Generex](
https://github.com/mifmif/Generex). I think the issue is in Generex, here is a quick sample:

```java
Generex generex = new Generex("^[A-Fa-f0-9]*$");

// Generate random String
String randomStr = generex.random();
System.out.println(randomStr);
```

This sample generates `^$` which confirms the issue is in Generex. This was already reported in https://github.com/mifmif/Generex/issues/41 and apparently rejected. So we will try to see how to "sanitize" the input on random beans side so the call to faker/generex is done with the right format expected by generex.

Kr,
Mahmoud

**Comment by benas** [comment]
Hi @tnmtechnologies 

We decided with @PascalSchumacher to fix the issue on RB side as it was rejected on the Generex side. The fix (in #319, many thanks to @PascalSchumacher for it) has been deployed to maven central in v3.8.0-SNAPSHOT. It would be great if you give it a try and let us know about it. I'm closing this issue for now, but feel free to add a comment here if you have any problem.

Kr,
Mahmoud
